core/src/main/java/org/apache/brooklyn/core/location/AbstractLocation.java

```diff
@@ -459,7 +459,7 @@ public ConfigBag getLocalBag() {
 
         @Override
         public void addToLocalBag(Map<String, ?> vals) {
-            configBag.putAll(vals);
+            configBag = ConfigBag.newInstanceExtending(configBag, vals);
```

Comment:
The exact answer here is, that the properties are set twice in `FixedListMachineProvisioningLocation#obtain`
1. when allflags is constructed and used in `provisionMore` call
2. after that the configuration of the object is updated for a second time in `updateMachineConfig` call